src/Data/Textile/Simulator.elm

```diff
@@ -508,15 +515,28 @@ computeSpinningStepWaste ({ inputs, lifeCycle } as simulator) =
                 |> (\inputMass ->
                         inputs.materials
                             |> List.map
-                                (\{ material, share } ->
+                                (\{ material, share, spinning } ->
                                     let
+                                        spinningProcess =
+                                            spinning
+                                                |> Maybe.withDefault (Material.getDefaultSpinning material.origin)
+
                                         processWaste =
-                                            material.spinningProcess
-                                                |> Maybe.map .waste
-                                                |> Maybe.withDefault (Mass.kilograms 0)
+                                            Material.wasteForSpinning spinningProcess
+
+                                        outputMaterialMass =
+                                            -- The output mass is the input mass of the next step
+                                            inputMass
+                                                |> Quantity.multiplyBy (Split.toFloat share)
+
+                                        inputMaterialMass =
+                                            -- Formula : inputMass - inputMass * waste = outputMass
+                                            -- => inputMass * (1 - waste) = outputMass
+                                            -- => inputMass = outputMass / (1 - waste)
+                                            Split.divideBy (Mass.inKilograms outputMaterialMass) (Split.complement processWaste)
```

Comment:
I know it's highly unlikely to ever happen but I'm slightly nervous with this giving `Infinity` when `Split.complement processWaste` is `0`. Well, that would output `Infinity` on the UI and we could live with it, and it will never happen anyway as no waste will ever be 100% so I understand we don't bother here.

Reply:
Yeah, I didn't really think of an alternative to be honest... Should I instead first validate that the waste is bigger than 0, or else return a default value? Which one? Or display an error message of some sorts? But it would make it a pain to deal with at this point in the code...